--- a/net/minecraft/item/crafting/MapExtendingRecipe.java
+++ b/net/minecraft/item/crafting/MapExtendingRecipe.java
@@ -9,6 +9,7 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.MapData;
 import net.minecraft.world.storage.MapDecoration;
+import org.bukkit.inventory.Recipe;
 
 public class MapExtendingRecipe extends ShapedRecipe {
    public MapExtendingRecipe(ResourceLocation p_i48164_1_) {
