--- a/net/minecraft/command/arguments/EntityOptions.java
+++ b/net/minecraft/command/arguments/EntityOptions.java
@@ -189,21 +189,21 @@
             });
             BiConsumer<Vector3d, List<? extends Entity>> biconsumer;
             switch(s) {
-            case "nearest":
-               biconsumer = EntitySelectorParser.field_197414_g;
-               break;
-            case "furthest":
-               biconsumer = EntitySelectorParser.field_197415_h;
-               break;
-            case "random":
-               biconsumer = EntitySelectorParser.field_197416_i;
-               break;
-            case "arbitrary":
-               biconsumer = EntitySelectorParser.field_197413_f;
-               break;
-            default:
-               p_197455_0_.func_197398_f().setCursor(i);
-               throw field_197475_h.createWithContext(p_197455_0_.func_197398_f(), s);
+               case "nearest":
+                  biconsumer = EntitySelectorParser.field_197414_g;
+                  break;
+               case "furthest":
+                  biconsumer = EntitySelectorParser.field_197415_h;
+                  break;
+               case "random":
+                  biconsumer = EntitySelectorParser.field_197416_i;
+                  break;
+               case "arbitrary":
+                  biconsumer = EntitySelectorParser.field_197413_f;
+                  break;
+               default:
+                  p_197455_0_.func_197398_f().setCursor(i);
+                  throw field_197475_h.createWithContext(p_197455_0_.func_197398_f(), s);
             }
 
             p_197455_0_.func_197376_a(biconsumer);
