--- a/net/minecraft/command/arguments/OperationArgument.java
+++ b/net/minecraft/command/arguments/OperationArgument.java
@@ -61,44 +_,44 @@
 
    private static OperationArgument.IIntOperation func_197182_b(String p_197182_0_) throws CommandSyntaxException {
       switch(p_197182_0_) {
-      case "=":
-         return (p_197174_0_, p_197174_1_) -> {
-            return p_197174_1_;
-         };
-      case "+=":
-         return (p_197176_0_, p_197176_1_) -> {
-            return p_197176_0_ + p_197176_1_;
-         };
-      case "-=":
-         return (p_197183_0_, p_197183_1_) -> {
-            return p_197183_0_ - p_197183_1_;
-         };
-      case "*=":
-         return (p_197173_0_, p_197173_1_) -> {
-            return p_197173_0_ * p_197173_1_;
-         };
-      case "/=":
-         return (p_197178_0_, p_197178_1_) -> {
-            if (p_197178_1_ == 0) {
-               throw field_197186_b.create();
-            } else {
-               return MathHelper.func_76137_a(p_197178_0_, p_197178_1_);
-            }
-         };
-      case "%=":
-         return (p_197181_0_, p_197181_1_) -> {
-            if (p_197181_1_ == 0) {
-               throw field_197186_b.create();
-            } else {
-               return MathHelper.func_180184_b(p_197181_0_, p_197181_1_);
-            }
-         };
-      case "<":
-         return Math::min;
-      case ">":
-         return Math::max;
-      default:
-         throw field_197185_a.create();
+         case "=":
+            return (p_197174_0_, p_197174_1_) -> {
+               return p_197174_1_;
+            };
+         case "+=":
+            return (p_197176_0_, p_197176_1_) -> {
+               return p_197176_0_ + p_197176_1_;
+            };
+         case "-=":
+            return (p_197183_0_, p_197183_1_) -> {
+               return p_197183_0_ - p_197183_1_;
+            };
+         case "*=":
+            return (p_197173_0_, p_197173_1_) -> {
+               return p_197173_0_ * p_197173_1_;
+            };
+         case "/=":
+            return (p_197178_0_, p_197178_1_) -> {
+               if (p_197178_1_ == 0) {
+                  throw field_197186_b.create();
+               } else {
+                  return MathHelper.func_76137_a(p_197178_0_, p_197178_1_);
+               }
+            };
+         case "%=":
+            return (p_197181_0_, p_197181_1_) -> {
+               if (p_197181_1_ == 0) {
+                  throw field_197186_b.create();
+               } else {
+                  return MathHelper.func_180184_b(p_197181_0_, p_197181_1_);
+               }
+            };
+         case "<":
+            return Math::min;
+         case ">":
+            return Math::max;
+         default:
+            throw field_197185_a.create();
       }
    }
 
