--- a/net/minecraft/util/text/filter/ChatFilterClient.java
+++ b/net/minecraft/util/text/filter/ChatFilterClient.java
@@ -148,8 +_,8 @@
       httpurlconnection.setRequestProperty("User-Agent", "Minecraft server" + SharedConstants.func_215069_a().getName());
 
       try (
-         OutputStreamWriter outputstreamwriter = new OutputStreamWriter(httpurlconnection.getOutputStream(), StandardCharsets.UTF_8);
-         JsonWriter jsonwriter = new JsonWriter(outputstreamwriter);
+          OutputStreamWriter outputstreamwriter = new OutputStreamWriter(httpurlconnection.getOutputStream(), StandardCharsets.UTF_8);
+          JsonWriter jsonwriter = new JsonWriter(outputstreamwriter);
       ) {
          Streams.write(p_244575_1_, jsonwriter);
       }
