--- a/net/minecraft/client/AbstractOption.java
+++ b/net/minecraft/client/AbstractOption.java
@@ -252,14 +252,14 @@
       }
    }, (p_216633_0_, p_216633_1_) -> {
       switch(p_216633_0_.field_238330_f_) {
-      case FAST:
-         p_216633_1_.func_241567_a_(Minecraft.func_71410_x().field_71466_p.func_238425_b_(field_241564_V_, 200));
-         break;
-      case FANCY:
-         p_216633_1_.func_241567_a_(Minecraft.func_71410_x().field_71466_p.func_238425_b_(field_241566_X_, 200));
-         break;
-      case FABULOUS:
-         p_216633_1_.func_241567_a_(Minecraft.func_71410_x().field_71466_p.func_238425_b_(field_241565_W_, 200));
+         case FAST:
+            p_216633_1_.func_241567_a_(Minecraft.func_71410_x().field_71466_p.func_238425_b_(field_241564_V_, 200));
+            break;
+         case FANCY:
+            p_216633_1_.func_241567_a_(Minecraft.func_71410_x().field_71466_p.func_238425_b_(field_241566_X_, 200));
+            break;
+         case FABULOUS:
+            p_216633_1_.func_241567_a_(Minecraft.func_71410_x().field_71466_p.func_238425_b_(field_241565_W_, 200));
       }
 
       IFormattableTextComponent iformattabletextcomponent = new TranslationTextComponent(p_216633_0_.field_238330_f_.func_238164_b_());
