--- a/net/minecraft/client/GameSettings.java
+++ b/net/minecraft/client/GameSettings.java
@@ -91,7 +91,6 @@
    public boolean field_181150_U = true;
    public AttackIndicatorStatus field_186716_M = AttackIndicatorStatus.CROSSHAIR;
    public TutorialSteps field_193631_S = TutorialSteps.MOVEMENT;
-   public boolean field_244601_E = false;
    public int field_205217_U = 2;
    public double field_208033_V = 1.0D;
    public boolean field_225307_E = true;
@@ -117,7 +116,6 @@
    public boolean field_228044_Y_;
    public boolean field_228045_Z_;
    public boolean field_230152_Z_;
-   public boolean field_244794_ae = true;
    public final KeyBinding field_74351_w = new KeyBinding("key.forward", 87, "key.categories.movement");
    public final KeyBinding field_74370_x = new KeyBinding("key.left", 65, "key.categories.movement");
    public final KeyBinding field_74368_y = new KeyBinding("key.back", 83, "key.categories.movement");
@@ -138,7 +136,6 @@
    public final KeyBinding field_74310_D = new KeyBinding("key.chat", 84, "key.categories.multiplayer");
    public final KeyBinding field_74321_H = new KeyBinding("key.playerlist", 258, "key.categories.multiplayer");
    public final KeyBinding field_74323_J = new KeyBinding("key.command", 47, "key.categories.multiplayer");
-   public final KeyBinding field_244602_au = new KeyBinding("key.socialInteractions", 80, "key.categories.multiplayer");
    public final KeyBinding field_151447_Z = new KeyBinding("key.screenshot", 291, "key.categories.misc");
    public final KeyBinding field_151457_aa = new KeyBinding("key.togglePerspective", 294, "key.categories.misc");
    public final KeyBinding field_151458_ab = new KeyBinding("key.smoothCamera", InputMappings.field_197958_a.func_197937_c(), "key.categories.misc");
@@ -148,7 +145,7 @@
    public final KeyBinding[] field_151456_ac = new KeyBinding[]{new KeyBinding("key.hotbar.1", 49, "key.categories.inventory"), new KeyBinding("key.hotbar.2", 50, "key.categories.inventory"), new KeyBinding("key.hotbar.3", 51, "key.categories.inventory"), new KeyBinding("key.hotbar.4", 52, "key.categories.inventory"), new KeyBinding("key.hotbar.5", 53, "key.categories.inventory"), new KeyBinding("key.hotbar.6", 54, "key.categories.inventory"), new KeyBinding("key.hotbar.7", 55, "key.categories.inventory"), new KeyBinding("key.hotbar.8", 56, "key.categories.inventory"), new KeyBinding("key.hotbar.9", 57, "key.categories.inventory")};
    public final KeyBinding field_193629_ap = new KeyBinding("key.saveToolbarActivator", 67, "key.categories.creative");
    public final KeyBinding field_193630_aq = new KeyBinding("key.loadToolbarActivator", 88, "key.categories.creative");
-   public KeyBinding[] field_74324_K = ArrayUtils.addAll(new KeyBinding[]{this.field_74312_F, this.field_74313_G, this.field_74351_w, this.field_74370_x, this.field_74368_y, this.field_74366_z, this.field_74314_A, this.field_228046_af_, this.field_151444_V, this.field_74316_C, this.field_151445_Q, this.field_74310_D, this.field_74321_H, this.field_74322_I, this.field_74323_J, this.field_244602_au, this.field_151447_Z, this.field_151457_aa, this.field_151458_ab, this.field_152395_am, this.field_178883_an, this.field_186718_X, this.field_193629_ap, this.field_193630_aq, this.field_194146_ao}, this.field_151456_ac);
+   public KeyBinding[] field_74324_K = ArrayUtils.addAll(new KeyBinding[]{this.field_74312_F, this.field_74313_G, this.field_74351_w, this.field_74370_x, this.field_74368_y, this.field_74366_z, this.field_74314_A, this.field_228046_af_, this.field_151444_V, this.field_74316_C, this.field_151445_Q, this.field_74310_D, this.field_74321_H, this.field_74322_I, this.field_74323_J, this.field_151447_Z, this.field_151457_aa, this.field_151458_ab, this.field_152395_am, this.field_178883_an, this.field_186718_X, this.field_193629_ap, this.field_193630_aq, this.field_194146_ao}, this.field_151456_ac);
    protected Minecraft field_74317_L;
    private final File field_74354_ai;
    public Difficulty field_74318_M = Difficulty.NORMAL;
@@ -170,6 +167,7 @@
    public boolean field_241568_aS_;
 
    public GameSettings(Minecraft p_i46326_1_, File p_i46326_2_) {
+      setForgeKeybindProperties();
       this.field_74317_L = p_i46326_1_;
       this.field_74354_ai = new File(p_i46326_2_, "options.txt");
       if (p_i46326_1_.func_147111_S() && Runtime.getRuntime().maxMemory() >= 1000000000L) {
@@ -515,21 +513,17 @@
                   this.field_230152_Z_ = "true".equals(s1);
                }
 
-               if ("hideMatchedNames".equals(s)) {
-                  this.field_244794_ae = "true".equals(s1);
-               }
-
-               if ("joinedFirstServer".equals(s)) {
-                  this.field_244601_E = "true".equals(s1);
-               }
-
                if ("syncChunkWrites".equals(s)) {
                   this.field_241568_aS_ = "true".equals(s1);
                }
 
                for(KeyBinding keybinding : this.field_74324_K) {
                   if (s.equals("key_" + keybinding.func_151464_g())) {
-                     keybinding.func_197979_b(InputMappings.func_197955_a(s1));
+                     if (s1.indexOf(':') != -1) {
+                        String[] pts = s1.split(":");
+                        keybinding.setKeyModifierAndCode(net.minecraftforge.client.settings.KeyModifier.valueFromString(pts[1]), InputMappings.func_197955_a(pts[0]));
+                     } else
+                        keybinding.setKeyModifierAndCode(net.minecraftforge.client.settings.KeyModifier.NONE, InputMappings.func_197955_a(s1));
                   }
                }
 
@@ -576,6 +570,7 @@
    }
 
    public void func_74303_b() {
+      if (net.minecraftforge.fml.client.ClientModLoader.isLoading()) return; //Don't save settings before mods add keybindigns and the like to prevent them from being deleted.
       try (PrintWriter printwriter = new PrintWriter(new OutputStreamWriter(new FileOutputStream(this.field_74354_ai), StandardCharsets.UTF_8))) {
          printwriter.println("version:" + SharedConstants.func_215069_a().getWorldVersion());
          printwriter.println("autoJump:" + AbstractOption.field_216719_z.func_216741_b(this));
@@ -656,12 +651,10 @@
          printwriter.println("rawMouseInput:" + AbstractOption.field_225302_l.func_216741_b(this));
          printwriter.println("glDebugVerbosity:" + this.field_209231_W);
          printwriter.println("skipMultiplayerWarning:" + this.field_230152_Z_);
-         printwriter.println("hideMatchedNames:" + this.field_244794_ae);
-         printwriter.println("joinedFirstServer:" + this.field_244601_E);
          printwriter.println("syncChunkWrites:" + this.field_241568_aS_);
 
          for(KeyBinding keybinding : this.field_74324_K) {
-            printwriter.println("key_" + keybinding.func_151464_g() + ":" + keybinding.func_197982_m());
+            printwriter.println("key_" + keybinding.func_151464_g() + ":" + keybinding.func_197982_m() + (keybinding.getKeyModifier() != net.minecraftforge.client.settings.KeyModifier.NONE ? ":" + keybinding.getKeyModifier() : ""));
          }
 
          for(SoundCategory soundcategory : SoundCategory.values()) {
@@ -760,6 +753,23 @@
       p_198017_1_.func_198985_a(set);
    }
 
+   private void setForgeKeybindProperties() {
+      net.minecraftforge.client.settings.KeyConflictContext inGame = net.minecraftforge.client.settings.KeyConflictContext.IN_GAME;
+      field_74351_w.setKeyConflictContext(inGame);
+      field_74370_x.setKeyConflictContext(inGame);
+      field_74368_y.setKeyConflictContext(inGame);
+      field_74366_z.setKeyConflictContext(inGame);
+      field_74314_A.setKeyConflictContext(inGame);
+      field_228046_af_.setKeyConflictContext(inGame);
+      field_151444_V.setKeyConflictContext(inGame);
+      field_74312_F.setKeyConflictContext(inGame);
+      field_74310_D.setKeyConflictContext(inGame);
+      field_74321_H.setKeyConflictContext(inGame);
+      field_74323_J.setKeyConflictContext(inGame);
+      field_151457_aa.setKeyConflictContext(inGame);
+      field_151458_ab.setKeyConflictContext(inGame);
+   }
+
    public PointOfView func_243230_g() {
       return this.field_243228_bb;
    }
