--- a/net/minecraft/client/tutorial/Tutorial.java
+++ b/net/minecraft/client/tutorial/Tutorial.java
@@ -1,11 +1,8 @@
 package net.minecraft.client.tutorial;
 
-import com.google.common.collect.Lists;
-import java.util.List;
 import javax.annotation.Nullable;
 import net.minecraft.block.BlockState;
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.gui.toasts.TutorialToast;
 import net.minecraft.client.world.ClientWorld;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.MovementInput;
@@ -23,7 +20,6 @@
    private final Minecraft field_193304_a;
    @Nullable
    private ITutorialStep field_193305_b;
-   private List<Tutorial.ToastTimeInfo> field_244696_c = Lists.newArrayList();
 
    public Tutorial(Minecraft p_i47578_1_) {
       this.field_193304_a = p_i47578_1_;
@@ -86,22 +82,7 @@
       this.field_193305_b = this.field_193304_a.field_71474_y.field_193631_S.func_193309_a(this);
    }
 
-   public void func_244698_a(TutorialToast p_244698_1_, int p_244698_2_) {
-      this.field_244696_c.add(new Tutorial.ToastTimeInfo(p_244698_1_, p_244698_2_));
-      this.field_193304_a.func_193033_an().func_192988_a(p_244698_1_);
-   }
-
-   public void func_244697_a(TutorialToast p_244697_1_) {
-      this.field_244696_c.removeIf((p_244699_1_) -> {
-         return p_244699_1_.field_244701_a == p_244697_1_;
-      });
-      p_244697_1_.func_193670_a();
-   }
-
    public void func_193303_d() {
-      this.field_244696_c.removeIf((p_244700_0_) -> {
-         return p_244700_0_.func_244704_a();
-      });
       if (this.field_193305_b != null) {
          if (this.field_193304_a.field_71441_e != null) {
             this.field_193305_b.func_193245_a();
@@ -135,26 +116,4 @@
    public static ITextComponent func_193291_a(String p_193291_0_) {
       return (new KeybindTextComponent("key." + p_193291_0_)).func_240699_a_(TextFormatting.BOLD);
    }
-
-   @OnlyIn(Dist.CLIENT)
-   static final class ToastTimeInfo {
-      private final TutorialToast field_244701_a;
-      private final int field_244702_b;
-      private int field_244703_c;
-
-      private ToastTimeInfo(TutorialToast p_i244518_1_, int p_i244518_2_) {
-         this.field_244701_a = p_i244518_1_;
-         this.field_244702_b = p_i244518_2_;
-      }
-
-      private boolean func_244704_a() {
-         this.field_244701_a.func_193669_a(Math.min((float)(++this.field_244703_c) / (float)this.field_244702_b, 1.0F));
-         if (this.field_244703_c > this.field_244702_b) {
-            this.field_244701_a.func_193670_a();
-            return true;
-         } else {
-            return false;
-         }
-      }
-   }
 }
