--- a/net/minecraft/tileentity/BlastFurnaceTileEntity.java
+++ b/net/minecraft/tileentity/BlastFurnaceTileEntity.java
@@ -8,6 +_,8 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TranslationTextComponent;
 
+import org.bukkit.craftbukkit.v1_16_R3.entity.CraftHumanEntity;
+
 public class BlastFurnaceTileEntity extends AbstractFurnaceTileEntity {
    public BlastFurnaceTileEntity() {
       super(TileEntityType.field_222484_B, IRecipeType.field_222151_c);
