--- a/net/minecraft/tileentity/BlastFurnaceTileEntity.java
+++ b/net/minecraft/tileentity/BlastFurnaceTileEntity.java
@@ -1,5 +_,6 @@
 package net.minecraft.tileentity;
 
+import java.util.List;
 import net.minecraft.entity.player.PlayerInventory;
 import net.minecraft.inventory.container.BlastFurnaceContainer;
 import net.minecraft.inventory.container.Container;
@@ -7,6 +_,10 @@
 import net.minecraft.item.crafting.IRecipeType;
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TranslationTextComponent;
+import org.bukkit.Location;
+import org.bukkit.craftbukkit.entity.CraftHumanEntity;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.inventory.InventoryHolder;
 
 public class BlastFurnaceTileEntity extends AbstractFurnaceTileEntity {
    public BlastFurnaceTileEntity() {
