--- a/net/minecraft/tileentity/FurnaceTileEntity.java
+++ b/net/minecraft/tileentity/FurnaceTileEntity.java
@@ -7,6 +_,8 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TranslationTextComponent;
 
+import org.bukkit.craftbukkit.v1_16_R3.entity.CraftHumanEntity;
+
 public class FurnaceTileEntity extends AbstractFurnaceTileEntity {
    public FurnaceTileEntity() {
       super(TileEntityType.field_200971_b, IRecipeType.field_222150_b);
