--- a/net/minecraft/tileentity/SmokerTileEntity.java
+++ b/net/minecraft/tileentity/SmokerTileEntity.java
@@ -8,6 +_,8 @@
 import net.minecraft.util.text.ITextComponent;
 import net.minecraft.util.text.TranslationTextComponent;
 
+import org.bukkit.craftbukkit.v1_16_R3.entity.CraftHumanEntity;
+
 public class SmokerTileEntity extends AbstractFurnaceTileEntity {
    public SmokerTileEntity() {
       super(TileEntityType.field_222483_A, IRecipeType.field_222152_d);
