--- a/com/mojang/realmsclient/dto/RealmsServer.java
+++ b/com/mojang/realmsclient/dto/RealmsServer.java
@@ -17,7 +17,6 @@
 import java.util.Objects;
 import java.util.Map.Entry;
 import net.minecraft.client.Minecraft;
-import net.minecraft.client.multiplayer.ServerData;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 import org.apache.commons.lang3.builder.EqualsBuilder;
@@ -274,10 +273,6 @@
       return this.field_230584_c_ + " (" + this.field_230590_i_.get(p_237696_1_).func_230787_a_(p_237696_1_) + ")";
    }
 
-   public ServerData func_244783_d(String p_244783_1_) {
-      return new ServerData(this.field_230584_c_, p_244783_1_, false);
-   }
-
    @OnlyIn(Dist.CLIENT)
    public static class ServerComparator implements Comparator<RealmsServer> {
       private final String field_223701_a;
