--- a/net/minecraft/network/login/client/CEncryptionResponsePacket.java
+++ b/net/minecraft/network/login/client/CEncryptionResponsePacket.java
@@ -7,7 +7,6 @@
 import net.minecraft.network.IPacket;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.IServerLoginNetHandler;
-import net.minecraft.util.CryptException;
 import net.minecraft.util.CryptManager;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
@@ -20,7 +19,7 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public CEncryptionResponsePacket(SecretKey p_i46851_1_, PublicKey p_i46851_2_, byte[] p_i46851_3_) throws CryptException {
+   public CEncryptionResponsePacket(SecretKey p_i46851_1_, PublicKey p_i46851_2_, byte[] p_i46851_3_) {
       this.field_149302_a = CryptManager.func_75894_a(p_i46851_2_, p_i46851_1_.getEncoded());
       this.field_149301_b = CryptManager.func_75894_a(p_i46851_2_, p_i46851_3_);
    }
@@ -39,11 +38,11 @@
       p_148833_1_.func_147315_a(this);
    }
 
-   public SecretKey func_149300_a(PrivateKey p_149300_1_) throws CryptException {
+   public SecretKey func_149300_a(PrivateKey p_149300_1_) {
       return CryptManager.func_75887_a(p_149300_1_, this.field_149302_a);
    }
 
-   public byte[] func_149299_b(PrivateKey p_149299_1_) throws CryptException {
-      return CryptManager.func_75889_b(p_149299_1_, this.field_149301_b);
+   public byte[] func_149299_b(PrivateKey p_149299_1_) {
+      return p_149299_1_ == null ? this.field_149301_b : CryptManager.func_75889_b(p_149299_1_, this.field_149301_b);
    }
 }
