--- a/net/minecraft/tags/ITagCollection.java
+++ b/net/minecraft/tags/ITagCollection.java
@@ -15,6 +_,8 @@
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.registry.DefaultedRegistry;
 import net.minecraft.util.registry.Registry;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
 
 public interface ITagCollection<T> {
    Map<ResourceLocation, ITag<T>> func_241833_a();
