--- a/net/minecraft/realms/action/ConnectingToRealmsAction.java
+++ b/net/minecraft/realms/action/ConnectingToRealmsAction.java
@@ -103,7 +103,7 @@
 
                         try {
                            Minecraft.func_71410_x().func_195541_I().func_217818_a(realmsserveraddress1.field_230602_b_, realmsserveraddress1.field_230603_c_).thenRun(() -> {
-                              this.func_224987_a(new RealmsLongRunningMcoTaskScreen(this.field_238117_d_, new ConnectedToRealmsAction(this.field_238117_d_, this.field_238116_c_, realmsserveraddress1)));
+                              this.func_224987_a(new RealmsLongRunningMcoTaskScreen(this.field_238117_d_, new ConnectedToRealmsAction(this.field_238117_d_, realmsserveraddress1)));
                            }).exceptionally(function);
                         } catch (Exception exception1) {
                            function.apply(exception1);
@@ -120,7 +120,7 @@
 
                }, RealmsLongConfirmationScreen.Type.Info, itextcomponent, itextcomponent1, true));
             } else {
-               this.func_224987_a(new RealmsLongRunningMcoTaskScreen(this.field_238117_d_, new ConnectedToRealmsAction(this.field_238117_d_, this.field_238116_c_, realmsserveraddress1)));
+               this.func_224987_a(new RealmsLongRunningMcoTaskScreen(this.field_238117_d_, new ConnectedToRealmsAction(this.field_238117_d_, realmsserveraddress1)));
             }
          } else {
             this.func_230434_a_(new TranslationTextComponent("mco.errorMessage.connectionFailure"));
