--- a/net/minecraft/client/gui/social/SocialInteractionsScreen.java
+++ b/net/minecraft/client/gui/social/SocialInteractionsScreen.java
@@ -41,7 +41,7 @@
    private FilterList field_244673_t;
    private TextFieldWidget field_244674_u;
    private String field_244675_v = "";
-   private SocialInteractionsScreen.Mode field_244676_w = SocialInteractionsScreen.Mode.ALL;
+   private Mode field_244676_w = Mode.ALL;
    private Button field_244677_x;
    private Button field_244678_y;
    private Button field_244760_E;
@@ -98,13 +98,13 @@
       int i1 = 64 + 16 * this.func_244690_l();
       int j1 = (this.field_230708_k_ - l) / 2;
       this.field_244677_x = this.func_230480_a_(new Button(j, 45, i, 20, field_244667_b, (p_244686_1_) -> {
-         this.func_244682_a(SocialInteractionsScreen.Mode.ALL);
+         this.func_244682_a(Mode.ALL);
       }));
       this.field_244678_y = this.func_230480_a_(new Button((j + k - i) / 2 + 1, 45, i, 20, field_244668_c, (p_244681_1_) -> {
-         this.func_244682_a(SocialInteractionsScreen.Mode.HIDDEN);
+         this.func_244682_a(Mode.HIDDEN);
       }));
       this.field_244760_E = this.func_230480_a_(new Button(k - i + 1, 45, i, 20, field_244762_p, (p_244769_1_) -> {
-         this.func_244682_a(SocialInteractionsScreen.Mode.BLOCKED);
+         this.func_244682_a(Mode.BLOCKED);
       }));
       this.field_244761_F = this.func_230480_a_(new Button(j1, i1, l, 20, field_244766_x, (p_244767_1_) -> {
          this.field_230706_i_.func_147108_a(new ConfirmOpenLinkScreen((p_244771_1_) -> {
@@ -133,7 +133,7 @@
       this.func_244682_a(this.field_244676_w);
    }
 
-   private void func_244682_a(SocialInteractionsScreen.Mode p_244682_1_) {
+   private void func_244682_a(Mode p_244682_1_) {
       this.field_244676_w = p_244682_1_;
       this.field_244677_x.func_238482_a_(field_244667_b);
       this.field_244678_y.func_238482_a_(field_244668_c);
@@ -162,9 +162,9 @@
       if (!this.field_244674_u.func_146179_b().isEmpty() && this.field_244673_t.func_244660_f() && !this.field_244674_u.func_230999_j_()) {
          NarratorChatListener.field_193643_a.func_216864_a(field_244764_u.getString());
       } else if (collection.isEmpty()) {
-         if (p_244682_1_ == SocialInteractionsScreen.Mode.HIDDEN) {
+         if (p_244682_1_ == Mode.HIDDEN) {
             NarratorChatListener.field_193643_a.func_216864_a(field_244672_s.getString());
-         } else if (p_244682_1_ == SocialInteractionsScreen.Mode.BLOCKED) {
+         } else if (p_244682_1_ == Mode.BLOCKED) {
             NarratorChatListener.field_193643_a.func_216864_a(field_244765_w.getString());
          }
       }
@@ -217,7 +217,7 @@
          this.field_244674_u.func_230430_a_(p_230430_1_, p_230430_2_, p_230430_3_, p_230430_4_);
       }
 
-      this.field_244761_F.field_230694_p_ = this.field_244676_w == SocialInteractionsScreen.Mode.BLOCKED;
+      this.field_244761_F.field_230694_p_ = this.field_244676_w == Mode.BLOCKED;
       super.func_230430_a_(p_230430_1_, p_230430_2_, p_230430_3_, p_230430_4_);
       if (this.field_244665_D != null) {
          this.field_244665_D.run();
