--- a/com/mojang/realmsclient/client/RealmsClient.java
+++ b/com/mojang/realmsclient/client/RealmsClient.java
@@ -2,7 +2,6 @@
 
 import com.mojang.realmsclient.dto.BackupList;
 import com.mojang.realmsclient.dto.Ops;
-import com.mojang.realmsclient.dto.PendingInvite;
 import com.mojang.realmsclient.dto.PendingInvitesList;
 import com.mojang.realmsclient.dto.PingResult;
 import com.mojang.realmsclient.dto.PlayerInfo;
@@ -21,9 +20,9 @@
 import com.mojang.realmsclient.exception.RealmsHttpException;
 import com.mojang.realmsclient.exception.RealmsServiceException;
 import com.mojang.realmsclient.exception.RetryCallException;
+import java.net.Proxy;
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.UUID;
 import javax.annotation.Nullable;
 import net.minecraft.client.Minecraft;
 import net.minecraft.realms.PersistenceSerializer;
@@ -40,7 +39,6 @@
    private static final Logger field_224946_c = LogManager.getLogger();
    private final String field_224947_d;
    private final String field_224948_e;
-   private final Minecraft field_244732_f;
    private static final PersistenceSerializer field_237691_f_ = new PersistenceSerializer();
 
    public static RealmsClient func_224911_a() {
@@ -63,7 +61,7 @@
          }
       }
 
-      return new RealmsClient(s1, s, minecraft);
+      return new RealmsClient(s1, s, minecraft.func_110437_J());
    }
 
    public static void func_224940_b() {
@@ -78,11 +76,10 @@
       field_224944_a = RealmsClient.Environment.LOCAL;
    }
 
-   public RealmsClient(String p_i244721_1_, String p_i244721_2_, Minecraft p_i244721_3_) {
-      this.field_224947_d = p_i244721_1_;
-      this.field_224948_e = p_i244721_2_;
-      this.field_244732_f = p_i244721_3_;
-      RealmsClientConfig.func_224896_a(p_i244721_3_.func_110437_J());
+   public RealmsClient(String p_i51790_1_, String p_i51790_2_, Proxy p_i51790_3_) {
+      this.field_224947_d = p_i51790_1_;
+      this.field_224948_e = p_i51790_2_;
+      RealmsClientConfig.func_224896_a(p_i51790_3_);
    }
 
    public RealmsServerList func_224902_e() throws RealmsServiceException {
@@ -243,26 +240,17 @@
    }
 
    public int func_224909_j() throws RealmsServiceException {
-      return this.func_224919_k().field_230569_a_.size();
+      String s = this.func_224926_c("invites/count/pending");
+      String s1 = this.func_224938_a(Request.func_224953_a(s));
+      return Integer.parseInt(s1);
    }
 
    public PendingInvitesList func_224919_k() throws RealmsServiceException {
       String s = this.func_224926_c("invites/pending");
       String s1 = this.func_224938_a(Request.func_224953_a(s));
-      PendingInvitesList pendinginviteslist = PendingInvitesList.func_230756_a_(s1);
-      pendinginviteslist.field_230569_a_.removeIf(this::func_244733_a);
-      return pendinginviteslist;
+      return PendingInvitesList.func_230756_a_(s1);
    }
 
-   private boolean func_244733_a(PendingInvite p_244733_1_) {
-      try {
-         UUID uuid = UUID.fromString(p_244733_1_.field_230566_d_);
-         return this.field_244732_f.func_244599_aA().func_244757_e(uuid);
-      } catch (IllegalArgumentException illegalargumentexception) {
-         return false;
-      }
-   }
-
    public void func_224901_a(String p_224901_1_) throws RealmsServiceException {
       String s = this.func_224926_c("invites" + "/accept/$INVITATION_ID".replace("$INVITATION_ID", p_224901_1_));
       this.func_224938_a(Request.func_224965_c(s, ""));
@@ -334,7 +322,10 @@
 
       try {
          int i = p_224938_1_.func_224958_b();
-         if (i != 503 && i != 277) {
+         if (i == 503) {
+            int j = p_224938_1_.func_224957_a();
+            throw new RetryCallException(j);
+         } else {
             String s = p_224938_1_.func_224963_c();
             if (i >= 200 && i < 300) {
                return s;
@@ -350,9 +341,6 @@
                field_224946_c.error("Realms error code: " + i + " message: " + s);
                throw new RealmsServiceException(i, s, i, "");
             }
-         } else {
-            int j = p_224938_1_.func_224957_a();
-            throw new RetryCallException(j, i);
          }
       } catch (RealmsHttpException realmshttpexception) {
          throw new RealmsServiceException(500, "Could not connect to Realms: " + realmshttpexception.getMessage(), -1, "");
