--- a/com/mojang/realmsclient/util/RealmsUtil.java
+++ b/com/mojang/realmsclient/util/RealmsUtil.java
@@ -12,6 +12,7 @@
 import com.mojang.util.UUIDTypeAdapter;
 import java.util.Date;
 import java.util.Map;
+import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 import net.minecraft.client.Minecraft;
 import net.minecraftforge.api.distmarker.Dist;
@@ -19,7 +20,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class RealmsUtil {
-   private static final YggdrasilAuthenticationService field_225195_b = new YggdrasilAuthenticationService(Minecraft.func_71410_x().func_110437_J());
+   private static final YggdrasilAuthenticationService field_225195_b = new YggdrasilAuthenticationService(Minecraft.func_71410_x().func_110437_J(), UUID.randomUUID().toString());
    private static final MinecraftSessionService field_225196_c = field_225195_b.createMinecraftSessionService();
    public static LoadingCache<String, GameProfile> field_225194_a = CacheBuilder.newBuilder().expireAfterWrite(60L, TimeUnit.MINUTES).build(new CacheLoader<String, GameProfile>() {
       public GameProfile load(String p_load_1_) throws Exception {
