--- a/net/minecraft/util/text/TextProcessing.java
+++ b/net/minecraft/util/text/TextProcessing.java
@@ -141,13 +141,4 @@
       });
       return stringbuilder.toString();
    }
-
-   public static String func_244782_a(ITextProperties p_244782_0_) {
-      StringBuilder stringbuilder = new StringBuilder();
-      func_238343_a_(p_244782_0_, Style.field_240709_b_, (p_244781_1_, p_244781_2_, p_244781_3_) -> {
-         stringbuilder.appendCodePoint(p_244781_3_);
-         return true;
-      });
-      return stringbuilder.toString();
-   }
 }
