--- a/net/minecraft/client/gui/screen/MultiplayerWarningScreen.java
+++ b/net/minecraft/client/gui/screen/MultiplayerWarningScreen.java
@@ -30,7 +30,7 @@
    protected void func_231160_c_() {
       super.func_231160_c_();
       this.field_243364_s = IBidiRenderer.func_243258_a(this.field_230712_o_, field_230158_c_, this.field_230708_k_ - 50);
-      int i = (this.field_243364_s.func_241862_a() + 1) * 9 * 2;
+      int i = (this.field_243364_s.func_241862_a() + 1) * 9;
       this.func_230480_a_(new Button(this.field_230708_k_ / 2 - 155, 100 + i, 150, 20, DialogTexts.field_240636_g_, (p_230165_1_) -> {
          if (this.field_230162_g_.func_212942_a()) {
             this.field_230706_i_.field_71474_y.field_230152_Z_ = true;
@@ -52,8 +52,8 @@
 
    public void func_230430_a_(MatrixStack p_230430_1_, int p_230430_2_, int p_230430_3_, float p_230430_4_) {
       this.func_231165_f_(0);
-      func_238475_b_(p_230430_1_, this.field_230712_o_, field_230157_b_, 25, 30, 16777215);
-      this.field_243364_s.func_241865_b(p_230430_1_, 25, 70, 9 * 2, 16777215);
+      func_238472_a_(p_230430_1_, this.field_230712_o_, field_230157_b_, this.field_230708_k_ / 2, 30, 16777215);
+      this.field_243364_s.func_241863_a(p_230430_1_, this.field_230708_k_ / 2, 70);
       super.func_230430_a_(p_230430_1_, p_230430_2_, p_230430_3_, p_230430_4_);
    }
 }
