--- a/net/minecraft/client/gui/screen/PackScreen.java
+++ b/net/minecraft/client/gui/screen/PackScreen.java
@@ -129,7 +129,7 @@
 
    private void func_238899_a_(net.minecraft.client.gui.widget.list.ResourcePackList p_238899_1_, Stream<PackLoadingManager.IPack> p_238899_2_) {
       p_238899_1_.func_231039_at__().clear();
-      p_238899_2_.forEach((p_238898_2_) -> {
+      p_238899_2_.filter(PackLoadingManager.IPack::notHidden).forEach((p_238898_2_) -> {
          p_238899_1_.func_231039_at__().add(new net.minecraft.client.gui.widget.list.ResourcePackList.ResourcePackEntry(this.field_230706_i_, p_238899_1_, this, p_238898_2_));
       });
    }
@@ -189,8 +189,8 @@
 
    private ResourceLocation func_243397_a(TextureManager p_243397_1_, ResourcePackInfo p_243397_2_) {
       try (
-         IResourcePack iresourcepack = p_243397_2_.func_195796_e();
-         InputStream inputstream = iresourcepack.func_195763_b("pack.png");
+          IResourcePack iresourcepack = p_243397_2_.func_195796_e();
+          InputStream inputstream = iresourcepack.func_195763_b("pack.png");
       ) {
          String s = p_243397_2_.func_195790_f();
          ResourceLocation resourcelocation = new ResourceLocation("minecraft", "pack/" + Util.func_244361_a(s, ResourceLocation::func_240909_b_) + "/" + Hashing.sha1().hashUnencodedChars(s) + "/icon");
