--- a/net/minecraft/client/gui/screen/WithNarratorSettingsScreen.java
+++ b/net/minecraft/client/gui/screen/WithNarratorSettingsScreen.java
@@ -31,7 +31,9 @@
       this.field_243315_q = new OptionsRowList(this.field_230706_i_, this.field_230708_k_, this.field_230709_l_, 32, this.field_230709_l_ - 32, 25);
       this.field_243315_q.func_214335_a(this.field_243313_c);
       this.field_230705_e_.add(this.field_243315_q);
-      this.func_244718_c();
+      this.func_230480_a_(new Button(this.field_230708_k_ / 2 - 100, this.field_230709_l_ - 27, 200, 20, DialogTexts.field_240632_c_, (p_243316_1_) -> {
+         this.field_230706_i_.func_147108_a(this.field_228182_a_);
+      }));
       this.field_243314_p = this.field_243315_q.func_243271_b(AbstractOption.field_216715_v);
       if (this.field_243314_p != null) {
          this.field_243314_p.field_230693_o_ = NarratorChatListener.field_193643_a.func_193640_a();
@@ -39,12 +41,6 @@
 
    }
 
-   protected void func_244718_c() {
-      this.func_230480_a_(new Button(this.field_230708_k_ / 2 - 100, this.field_230709_l_ - 27, 200, 20, DialogTexts.field_240632_c_, (p_243316_1_) -> {
-         this.field_230706_i_.func_147108_a(this.field_228182_a_);
-      }));
-   }
-
    public void func_230430_a_(MatrixStack p_230430_1_, int p_230430_2_, int p_230430_3_, float p_230430_4_) {
       this.func_230446_a_(p_230430_1_);
       this.field_243315_q.func_230430_a_(p_230430_1_, p_230430_2_, p_230430_3_, p_230430_4_);
