--- a/net/minecraft/network/play/server/SEntityPropertiesPacket.java
+++ b/net/minecraft/network/play/server/SEntityPropertiesPacket.java
@@ -13,6 +_,8 @@
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.registry.Registry;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class SEntityPropertiesPacket implements IPacket<IClientPlayNetHandler> {
    private int field_149445_a;
@@ -73,6 +_,7 @@
       p_148833_1_.func_147290_a(this);
    }
 
+   @OnlyIn(Dist.CLIENT)
    public int func_149442_c() {
       return this.field_149445_a;
    }
