--- a/net/minecraft/resources/IPackNameDecorator.java
+++ b/net/minecraft/resources/IPackNameDecorator.java
@@ -1,7 +1,6 @@
 package net.minecraft.resources;
 
 import net.minecraft.util.text.ITextComponent;
-import net.minecraft.util.text.TextFormatting;
 import net.minecraft.util.text.TranslationTextComponent;
 
 public interface IPackNameDecorator {
@@ -21,7 +20,7 @@
    static IPackNameDecorator func_232630_a_(String p_232630_0_) {
       ITextComponent itextcomponent = new TranslationTextComponent(p_232630_0_);
       return (p_232632_1_) -> {
-         return (new TranslationTextComponent("pack.nameAndSource", p_232632_1_, itextcomponent)).func_240699_a_(TextFormatting.GRAY);
+         return new TranslationTextComponent("pack.nameAndSource", p_232632_1_, itextcomponent);
       };
    }
 }
