--- a/net/minecraft/client/gui/screen/inventory/AnvilScreen.java
+++ b/net/minecraft/client/gui/screen/inventory/AnvilScreen.java
@@ -27,11 +27,6 @@
       this.field_238742_p_ = 60;
    }
 
-   public void func_231023_e_() {
-      super.func_231023_e_();
-      this.field_147091_w.func_146178_a();
-   }
-
    protected void func_230453_j_() {
       this.field_230706_i_.field_195559_v.func_197967_a(true);
       int i = (this.field_230708_k_ - this.field_146999_f) / 2;
