--- a/net/minecraft/client/gui/chat/NarratorChatListener.java
+++ b/net/minecraft/client/gui/chat/NarratorChatListener.java
@@ -24,19 +24,21 @@
    private final Narrator field_192580_a = Narrator.getNarrator();
 
    public void func_192576_a(ChatType p_192576_1_, ITextComponent p_192576_2_, UUID p_192576_3_) {
-      NarratorStatus narratorstatus = func_223131_d();
-      if (narratorstatus != NarratorStatus.OFF && this.field_192580_a.active()) {
-         if (narratorstatus == NarratorStatus.ALL || narratorstatus == NarratorStatus.CHAT && p_192576_1_ == ChatType.CHAT || narratorstatus == NarratorStatus.SYSTEM && p_192576_1_ == ChatType.SYSTEM) {
-            ITextComponent itextcomponent;
-            if (p_192576_2_ instanceof TranslationTextComponent && "chat.type.text".equals(((TranslationTextComponent)p_192576_2_).func_150268_i())) {
-               itextcomponent = new TranslationTextComponent("chat.type.text.narrate", ((TranslationTextComponent)p_192576_2_).func_150271_j());
-            } else {
-               itextcomponent = p_192576_2_;
+      if (!Minecraft.func_71410_x().func_238198_a_(p_192576_3_)) {
+         NarratorStatus narratorstatus = func_223131_d();
+         if (narratorstatus != NarratorStatus.OFF && this.field_192580_a.active()) {
+            if (narratorstatus == NarratorStatus.ALL || narratorstatus == NarratorStatus.CHAT && p_192576_1_ == ChatType.CHAT || narratorstatus == NarratorStatus.SYSTEM && p_192576_1_ == ChatType.SYSTEM) {
+               ITextComponent itextcomponent;
+               if (p_192576_2_ instanceof TranslationTextComponent && "chat.type.text".equals(((TranslationTextComponent)p_192576_2_).func_150268_i())) {
+                  itextcomponent = new TranslationTextComponent("chat.type.text.narrate", ((TranslationTextComponent)p_192576_2_).func_150271_j());
+               } else {
+                  itextcomponent = p_192576_2_;
+               }
+
+               this.func_216866_a(p_192576_1_.func_218690_b(), itextcomponent.getString());
             }
 
-            this.func_216866_a(p_192576_1_.func_218690_b(), itextcomponent.getString());
          }
-
       }
    }
 
