--- a/net/minecraft/client/gui/chat/OverlayChatListener.java
+++ b/net/minecraft/client/gui/chat/OverlayChatListener.java
@@ -16,6 +16,8 @@
    }
 
    public void func_192576_a(ChatType p_192576_1_, ITextComponent p_192576_2_, UUID p_192576_3_) {
-      this.field_192577_a.field_71456_v.func_175188_a(p_192576_2_, false);
+      if (!this.field_192577_a.func_238198_a_(p_192576_3_)) {
+         this.field_192577_a.field_71456_v.func_175188_a(p_192576_2_, false);
+      }
    }
 }
