--- a/net/minecraft/network/login/server/SEncryptionRequestPacket.java
+++ b/net/minecraft/network/login/server/SEncryptionRequestPacket.java
@@ -5,34 +5,33 @@
 import net.minecraft.client.network.login.IClientLoginNetHandler;
 import net.minecraft.network.IPacket;
 import net.minecraft.network.PacketBuffer;
-import net.minecraft.util.CryptException;
 import net.minecraft.util.CryptManager;
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 public class SEncryptionRequestPacket implements IPacket<IClientLoginNetHandler> {
    private String field_149612_a;
-   private byte[] field_149610_b;
+   private PublicKey field_149610_b;
    private byte[] field_149611_c;
 
    public SEncryptionRequestPacket() {
    }
 
-   public SEncryptionRequestPacket(String p_i244727_1_, byte[] p_i244727_2_, byte[] p_i244727_3_) {
-      this.field_149612_a = p_i244727_1_;
-      this.field_149610_b = p_i244727_2_;
-      this.field_149611_c = p_i244727_3_;
+   public SEncryptionRequestPacket(String p_i46855_1_, PublicKey p_i46855_2_, byte[] p_i46855_3_) {
+      this.field_149612_a = p_i46855_1_;
+      this.field_149610_b = p_i46855_2_;
+      this.field_149611_c = p_i46855_3_;
    }
 
    public void func_148837_a(PacketBuffer p_148837_1_) throws IOException {
       this.field_149612_a = p_148837_1_.func_150789_c(20);
-      this.field_149610_b = p_148837_1_.func_179251_a();
+      this.field_149610_b = CryptManager.func_75896_a(p_148837_1_.func_179251_a());
       this.field_149611_c = p_148837_1_.func_179251_a();
    }
 
    public void func_148840_b(PacketBuffer p_148840_1_) throws IOException {
       p_148840_1_.func_180714_a(this.field_149612_a);
-      p_148840_1_.func_179250_a(this.field_149610_b);
+      p_148840_1_.func_179250_a(this.field_149610_b.getEncoded());
       p_148840_1_.func_179250_a(this.field_149611_c);
    }
 
@@ -46,8 +45,8 @@
    }
 
    @OnlyIn(Dist.CLIENT)
-   public PublicKey func_149608_d() throws CryptException {
-      return CryptManager.func_75896_a(this.field_149610_b);
+   public PublicKey func_149608_d() {
+      return this.field_149610_b;
    }
 
    @OnlyIn(Dist.CLIENT)
